test/manifest_reader_test.cc

```diff
@@ -154,51 +173,64 @@ class ManifestReaderV2Test : public TempFileTestBase {
+                       .split_offsets = {4},
+                       .equality_ids = {},
+                       .sort_order_id = 0,
+                       .first_row_id = std::nullopt,
+                       .referenced_data_file = std::nullopt,
+                       .content_offset = std::nullopt,
+                       .content_size_in_bytes = std::nullopt};
+
+    if (partition_spec_id.has_value()) {
+      data_file.partition_spec_id = partition_spec_id.value();
+    }
+
     manifest_entries.emplace_back(
         ManifestEntry{.status = ManifestStatus::kAdded,
                       .snapshot_id = 679879563479918846LL,
-                      .sequence_number = std::nullopt,
-                      .file_sequence_number = std::nullopt,
-                      .data_file = std::make_shared<DataFile>(
-                          DataFile{.file_path = test_dir_prefix + paths[0],
-                                   .file_format = FileFormatType::kParquet,
-                                   .record_count = record_counts[0],
-                                   .file_size_in_bytes = file_sizes[0],
-                                   .column_sizes = {{1, 56}, {2, 73}, {3, 66}, {4, 67}},
-                                   .value_counts = {{1, 4}, {2, 4}, {3, 4}, {4, 4}},
-                                   .null_value_counts = {{1, 0}, {2, 0}, {3, 0}, {4, 0}},
-                                   .nan_value_counts = {{4, 0}},
-                                   .lower_bounds = lower_bounds[0],
-                                   .upper_bounds = upper_bounds[0],
-                                   .key_metadata = {},
-                                   .split_offsets = {4},
-                                   .equality_ids = {},
-                                   .sort_order_id = 0,
-                                   .first_row_id = std::nullopt,
-                                   .referenced_data_file = std::nullopt,
-                                   .content_offset = std::nullopt,
-                                   .content_size_in_bytes = std::nullopt})});
+                      .sequence_number = sequence_number,
+                      .file_sequence_number = sequence_number,
+                      .data_file = std::make_shared<DataFile>(data_file)});
     return manifest_entries;
   }
 
-  std::shared_ptr<::arrow::fs::LocalFileSystem> local_fs_;
-  std::shared_ptr<FileIO> file_io_;
-};
-
-TEST_F(ManifestReaderV2Test, V2NonPartitionedBasicTest) {
-  std::string path = GetResourcePath("2ddf1bc9-830b-4015-aced-c060df36f150-m0.avro");
+  std::vector<ManifestEntry> PrepareNonPartitionedTestData() {
+    return CreateV2TestData();
+  }
 
-  auto manifest_reader_result = ManifestReader::MakeReader(path, file_io_, nullptr);
-  ASSERT_EQ(manifest_reader_result.has_value(), true)
-      << manifest_reader_result.error().message;
+  std::vector<ManifestEntry> PrepareMetadataInheritanceTestData() {
+    return CreateV2TestData(15, 12);
```

Comment:
```suggestion
    return CreateV2TestData(/*sequence_number=*/15, /*partition_spec_id*/12);
```

For better readability.